### PR TITLE
fix(wmg): Update secondary filter options

### PR DESCRIPTION
**Pull request by tihuan**
### Reviewers
**Functional:** 
@seve 
**Readability:** 

---

## Changes
Update `useFilterDimensions()` hook to include an option to `useSecondaryFilters` or not. Default to `true`, which will tell BE to return filtered secondary options. If `useSecondaryFilters` is `false`, BE will return all available secondary filter options, which is what we want for the Filters panel to show

As seen in the screenshot below, the left filter panel still shows all available datasets, but the right info panel's `Source Data` section only shows the filtered 2 datasets 

Screen Shot 2022-04-04 at 2 41 59 PM

### CODE REVIEW

File: frontend/src/common/queries/wheresMyGene.ts [hunk 1]
```diff
@@ -477,7 +485,17 @@ function aggregateIdLabels(items: { [id: string]: string }[]): {
...
 }
 
-function useWMGQueryRequestBody() {
+const EMPTY_FILTERS = {
+  datasets: undefined,
+  developmentStages: undefined,
+  diseases: undefined,
+  ethnicities: undefined,
+  sexes: undefined,
+};
```
Comment (seve) [inline frontend/src/common/queries/wheresMyGene.ts:494]: Is there a type we can assign to this object to make sure it doesn't break if we add any filters in the future?

File: frontend/src/common/queries/wheresMyGene.ts [hunk 2]
```diff
@@ -488,7 +506,7 @@ function useWMGQueryRequestBody() {
   const { data } = usePrimaryFilterDimensions();
 
   const { datasets, developmentStages, diseases, ethnicities, sexes } =
-    selectedFilters;
+    includeAllFilterOptions ? EMPTY_FILTERS : selectedFilters;
```
Comment (seve) [inline frontend/src/common/queries/wheresMyGene.ts:509]: Can you comment explaining this ternary? Seems like it's the crux of these changes, but I don't seem to understand the implications

**Review by seve — CHANGES_REQUESTED** [review]

File: frontend/src/common/queries/wheresMyGene.ts (hunk 1 above)
Comment (tihuan) [reply frontend/src/common/queries/wheresMyGene.ts:494]: Ooh yes, really good point. Thank you! Done!

File: frontend/src/common/queries/wheresMyGene.ts (hunk 2 above)
Comment (tihuan) [reply frontend/src/common/queries/wheresMyGene.ts:509]: Good call out thank you! Done!

**Comment by tihuan** [comment]
PTAL again @seve 🙏 Thanks for the great feedback!

**Review by seve — APPROVED** [review]

**Pull request merged**